regulation-worker/internal/delete/batch/batch_test.go

```diff
@@ -57,12 +71,67 @@ func TestBatchDelete(t *testing.T) {
 			},
 		},
 	}
-	bm := batch.BatchManager{}
+	bm := batch.BatchManager{
+		FMFactory: mockFileManagerFactory{},
+	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			t.Skip()
 			status := bm.Delete(ctx, tt.job, tt.dest.Config, tt.dest.Name)
 			require.Equal(t, model.JobStatusComplete, status)
+
+			searchDir := mockBucketLocation
+			var cleanedFilesList []string
+			err := filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
+				if regexRequiredSuffix.Match([]byte(path)) {
+					cleanedFilesList = append(cleanedFilesList, path)
+				}
+				return nil
+			})
+			if err != nil {
+				panic(err)
+			}
+			sort.Strings(cleanedFilesList)
+
+			searchDir = "./goldenFile"
+			var goldenFilesList []string
+			err = filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
+				if regexRequiredSuffix.Match([]byte(path)) {
+					goldenFilesList = append(goldenFilesList, path)
+				}
+				return nil
+			})
+			if err != nil {
+				panic(err)
+			}
+			sort.Strings(goldenFilesList)
+			require.Equal(t, len(goldenFilesList), len(cleanedFilesList), "actual number of files in destination bucket different than expected")
+			for i := 0; i < len(goldenFilesList); i++ {
+				goldenFilePtr, err := os.OpenFile(goldenFilesList[i], os.O_RDWR, 0644)
```

Comment:
Any reason for using `OpenFile` ? We can simplify things by:
```suggestion
				goldenFilePtr, err := os.Open(goldenFilesList[i])
```